legend-pure-m3-core/src/main/java/org/finos/legend/pure/m3/serialization/filesystem/usercodestorage/CodeStorageTools.java

```diff
@@ -163,4 +163,18 @@ public static String joinPaths(String... paths)
 
         return new String(chars, 0, length);
     }
+
+    public static String getRepoNameForPath(String path)
+    {
+        if (!path.isEmpty() && (path.charAt(0) == '/'))
+        {
+            int index = path.indexOf('/', 1);
+            if (index != -1)
+            {
+                return path.substring(1, index);
+            }
+            return path.substring(1);
```

Comment:
This is not correct. If there is no subsequent '/', then the repo should be null. For example, "/welcome.pure" should result in null being returned.